src/main.cpp

```diff
@@ -4840,14 +4848,20 @@ bool ProcessBlock(CNode* pfrom, CBlock* pblock, bool generated_by_me)
              mi != mapOrphanBlocksByPrev.upper_bound(hashPrev);
              ++mi)
         {
-            CBlock* pblockOrphan = (*mi).second;
+            CBlock* pblockOrphan = mi->second;
             if (pblockOrphan->AcceptBlock(generated_by_me))
                 vWorkQueue.push_back(pblockOrphan->GetHash());
             mapOrphanBlocks.erase(pblockOrphan->GetHash());
             setStakeSeenOrphan.erase(pblockOrphan->GetProofOfStake());
             delete pblockOrphan;
         }
+
+        auto it = mapOrphanBlocksByPrev.begin();
+        while((it = mapOrphanBlocksByPrev.find(hashPrev)) != mapOrphanBlocksByPrev.end()) {
+            delete it->second;
```

Comment:
This deletion already happens the the `mapOrphanBlocksByPrev` is iterated on in the loop above.